common/src/main/java/org/dash/wallet/common/ui/dialogs/AdaptiveDialog.kt

```diff
@@ -63,6 +61,27 @@ class AdaptiveDialog(@LayoutRes private val layout: Int): DialogFragment() {
             )
         }
 
+        suspend fun withProgress(massage: String, activity: FragmentActivity, action: suspend () -> Unit) {
+            val dialog = progress(massage)
+            dialog.show(activity) { }
+            action.invoke()
+            dialog.dismiss()
+        }
+
+        @JvmStatic
+        fun progress(
+            message: String
+        ): AdaptiveDialog {
+            return custom(
+                R.layout.dialog_progress,
+                null,
+                null,
+                message,
+                "",
+                null
+            ).apply { isCancelable = false }
+        }
+
```

Comment:
In DashPay, the FancyAlertDialog does have some progress related feature.  Not sure if that was your inspiration here, but if we were to replace `FancyAlertDialog` with `AdaptiveDialog`, we would need these progress features in the DashPay branch.

Reply:
Progress in the `FancyAlertDialog` used quite a lot in the Coinbase branch. I figured we might want to put it in the `AdaptiveDialog` as well since `FancyAlertDialog` is being faded out. I suppose we can do the same for DashPay.